### ISSUE TITLE
[v3] Desk Tool crashes when you click on a tab (group) on one singleton document then navigate to another.

### ISSUE THREAD

**Issue by LouieRichardson99** [sanity-io/sanity#4000]
**Describe the bug**

The desk tool crashes due to the groups not resetting when you navigate to another singleton document.  Therefore, if I click on a group ('seo') in one document and navigate to another, it will look for that group ('seo') on the newly loaded document, but if there isn't a group with that name, it will crash.

Here's the error message I receive:
```
Error: Cannot read properties of undefined (reading 'name')
TypeError: Cannot read properties of undefined (reading 'name')
    at isFieldEnabledByGroupFilter (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:144988:20)
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:145140:36
    at Array.flatMap (<anonymous>)
    at prepareObjectInputState (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:145136:43)
    at prepareFormState (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:145251:10)
    at http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:145261:18
    at updateMemo (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:27020:27)
    at Object.useMemo (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:27536:24)
    at useMemo (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-U4BEXJXG.js?v=ad169d72:1047:29)
    at useFormState (http://localhost:3333/node_modules/.sanity/vite/deps/chunk-5YGJSBIU.js?v=ad169d72:145260:37)
```

Here's a video of the bug in action:
https://user-images.githubusercontent.com/60019261/208629608-6eea1972-366c-4249-ab38-13b205fd4d2d.mov

**To Reproduce**

1. Add the following code to your `sanity.config.js`:
```
import { defineConfig } from 'sanity'
import { deskTool } from 'sanity/desk'
import { visionTool } from '@sanity/vision'

export default defineConfig({
  name: 'default',
  title: 'Test',
  projectId: 'xxx',
  dataset: 'production',
  plugins: [
    deskTool({
      structure: (S) =>
        S.list()
          .title('Base')
          .items([
            S.documentListItem()
              .title('Test Page 1')
              .id('test1')
              .schemaType('test1'),
            S.documentListItem()
              .title('Test Page 2')
              .id('test2')
              .schemaType('test2')
          ])
    }),
    visionTool()
  ],
  schema: {
    types: [
      {
        name: 'test1',
        title: 'Test 1',
        type: 'document',
        groups: [{ name: 'seo', title: 'SEO' }],
        fields: [
          {
            name: 'title',
            title: 'Title',
            type: 'string'
          },
          {
            name: 'metaTitle',
            title: 'Meta Title',
            type: 'string',
            group: 'seo'
          }
        ]
      },
      {
        name: 'test2',
        title: 'Test 2',
        type: 'document',
        fields: [
          {
            name: 'title',
            title: 'Title',
            type: 'string'
          }
        ]
      }
    ]
  }
})
```

2. Once you've added in the code, replace the projectId so you can preview the desk tool.
3. Go to the desk tool, and navigate to 'Test Page 1'.
4. Click the 'SEO' tab (group).
5. Navigate to 'Test Page 2'.

**Expected behavior**

I expect it to default back to the normal group when I click on another document.

**Which versions of Sanity are you using?**

Run `sanity versions` in the terminal and copy-paste the result here.

```
@sanity/cli (global)          3.0.6 (up to date)
@sanity/eslint-config-studio  2.0.1 (up to date)
@sanity/vision                3.0.6 (up to date)
sanity                        3.0.6 (up to date)
```

**What operating system are you using?**

macOS, version 13.0.1

**Which versions of Node.js / npm are you running?**

```
9.2.0
v16.15.1
```

**Additional context**

This only seems to happen when I lay out the desk tool using the `S.documentListItem()` instead of the `S.listItem()`. Here's my schema for what my `deskTool` looks like:

```
deskTool({
      structure: (S) =>
        S.list()
          .title('Base')
          .items([
            S.listItem()
              .title('Navigation')
              .icon(MenuIcon)
              .child(
                S.list()
                  .title('Navigation Types')
                  .items([
                    S.documentListItem().title('Header').id('header').schemaType('header'),
                    S.documentListItem().title('Footer').id('footer').schemaType('footer'),
                  ])
              ),
            S.documentListItem()
              .title('Socials')
              .icon(LinkIcon)
              .id('socials')
              .schemaType('socials'),
            S.divider(),
            S.documentListItem()
              .title('Homepage')
              .icon(HomeIcon)
              .id('homepage')
              .schemaType('homepage'),
            S.listItem()
              .title('Pages')
              .icon(DocumentsIcon)
              .child(S.documentList().title('All Pages').filter('_type == "page"')),
          ]),
    }),
```

[Link to GitHub repo](https://github.com/LouieRichardson99/visualise-digital)

**Comment by rylanharper** [comment]
Can confirm. This is happening with Sanity `^3.1.0` as well. Not a breaking change, but definitely needs to be fixed asap

**Comment by LouieRichardson99** [comment]
@rylanharper I'm surprised other people haven't spotted this issue too since I'm sure a lot more people are structuring their schemas like this.

**Comment by rylanharper** [comment]
Bumping this issue 🚀

**Comment by bjoerge** [comment]
This should be addressed by https://github.com/sanity-io/sanity/pull/4050 which is scheduled for release in the next couple of days. Thank you for your patience!

**Comment by bjoerge** [comment]
Fixed in [v3.2.0](https://github.com/sanity-io/sanity/releases/tag/v3.2.0)
